### ISSUE TITLE
SVG resources don't seem to get rendered

### ISSUE THREAD

**Issue by informeti** [copperspice/copperspice#231]
It has come to our attention, that we can't use SVG icons in our MFC+CopperSpice project.

Interestingly, this code works fine when build with Qt 5 and qmake, leading us to believe that something must be different with CopperSpice.

With `btnExcelExport` being a `QPushButton*`, the following code leaves the button empty, without icon:
```c++
    ui().btnExcelExport->setIcon(QIcon(INQSTR(":/image/test.svg")));
```

...while the following code does render an icon:
```c++
    ui().btnExcelExport->setIcon(QIcon(INQSTR(":/image/baseline_ios_share_white_18dp.png")));
```

We've tried a few icons from https://fonts.google.com/icons?selected=Material+Icons as well as some minimal SVGs for testing purposes:

```xml
<svg width="40" height="11">
  <rect width="30" height="10" style="fill:rgb(0,0,255);stroke-width:3;stroke:rgb(0,0,0)" />
  Sorry, your browser does not support inline SVG.  
</svg>
```

```xml
<svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"/></svg>
```

We also have the issue with SVGs being using in style sheets, although I haven't tested if that particular use case works in Qt 5.

Any suggestions on how we could proceed?

We're using CopperSpice 1.7.3 built into a NuGet package with CMake & Ninja.

The actual project is an MFC project in which we integrate CopperSpice using the QtWinMigrate classes. The project is build with VS2019.
We link againts CsCore, CsGui, and - since this problem came to our attention - also against CsSvg.

Many thanks and kind regards!

**Comment by bgeller** [comment]
We verified that SVG images render in our KitchenSink demo program. There is always the possibility were are doing things differently.  I aim is to find out what that is make any necessary corrections to our code base.

>We're using CopperSpice 1.7.3 built into a NuGet package with CMake & Ninja

Would it be possible for you to try our binary build of CS 1.7.4 in a small stand alone program with no other non-CS libraries? These files are available on github and our download site. 

Once we can narrow down the issue I am sure we can resolve this problem. Thanks so much for reaching out to us.

Barbara

**Comment by informeti** [comment]
Thank you very much for your assistance.

I set up a minimal project which uses CopperSpice 1.7.4 (the x64 MSVC version) and the following code:

```c++
#include <QtCore/QtCore>
#include <QtGui/QtGui>

using namespace std;

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);

    QMainWindow d;
    d.resize(400, 300);

    QPushButton b(&d);
    b.setText(QString::fromUtf8("QPushButton"));

    // This works:
    //b.setIcon(QIcon(QString::fromUtf8(":/image/image/baseline_ios_share_black_18dp.png")));
    
    // This doesn't work:
    b.setIcon(QIcon(QString::fromUtf8(":/image/image/circle_black_24dp.svg")));

    d.show();

    return app.exec();
}
```

When using the PNG (https://fonts.google.com/icons?selected=Material%20Icons%3Aios_share%3A), the icon appears:
![image](https://user-images.githubusercontent.com/105800754/169831429-7dbbe564-0a4a-481b-a887-d293cfec7003.png)

When using the SVG (https://fonts.google.com/icons?selected=Material%20Icons%3Acircle%3A), it doesn't:
![image](https://user-images.githubusercontent.com/105800754/169831348-6e67e9ef-6b40-4d03-879d-23994ba043c3.png)

The application output in QtCreator doesn't show anything besides `QWindowsContext::setProcessDpiAwareness(): Dpi = 2`.

As for building, I used the following directory layout and a single CMakeLists.txt:
```
├── CMakeLists.txt
├── CMakeLists.txt.user
├── Makefile
├── app.qrc
├── image
│   ├── baseline_ios_share_black_18dp.png
│   └── circle_black_24dp.svg
├── main.cpp
├── mainwindow.cpp
├── mainwindow.h
└── mainwindow.ui
```

The CMakeLists.txt file looks like this (it's taken from the two CMakeLists.txt examples from the docs):
```cmake
cmake_minimum_required(VERSION 3.16 FATAL_ERROR)

project(copperspice-test)

include(CheckCXXCompilerFlag)
include(CheckCXXSourceCompiles)
include(CheckIncludeFile)
include(CheckIncludeFiles)
include(CheckTypeSize)

find_package(CopperSpice REQUIRED)

set(PACKAGE "copperspice-test")
set(PACKAGE_NAME "copperspice-test")

set(CMAKE_INCLUDE_CURRENT_DIR ON)
set(CMAKE_INCLUDE_DIRECTORIES_BEFORE ON)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_STANDARD 17)

# location for binary files
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)

list(APPEND app_sources
   ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/mainwindow.ui

   qrc_app.cpp
)

# run rcc to generate qrc output
COPPERSPICE_RESOURCES(
   ${CMAKE_CURRENT_SOURCE_DIR}/app.qrc
)

# run uic to generate ui_*.h source files
COPPERSPICE_RESOURCES(
   ${CMAKE_CURRENT_SOURCE_DIR}/mainwindow.ui
)

# qm files will be placed in the resource folder
set(TS_OUTPUT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/resources)

add_executable(copperspice-test
   ${app_sources}
)

target_include_directories(copperspice-test
   PRIVATE
   ${CMAKE_SOURCE_DIR}/src
)

target_link_libraries(copperspice-test
   CopperSpice::CsCore
   CopperSpice::CsGui
)

if(CMAKE_SYSTEM_NAME MATCHES "Windows")
   # disable console on windows
   target_link_libraries(copperspice-test -lnetapi32 -lmpr -mwindows)
endif()

install(TARGETS copperspice-test DESTINATION .)
```

**Comment by agserm** [comment]
We have been able to reproduce the behavior you reported with the SVG file and someone on our team will investigate.

**Comment by bgeller** [comment]
We have enabled the code to build the SVG plugin and everything has passed our CI testing. Our sample now displays the image.  This will be pushed to github shortly.

**Comment by informeti** [comment]
That sounds amazing, thanks a lot!

Do you have any idea when 1.7.5 will be released approximately, or would you recommend we rather backport the respective commit(s) into our build?

**Comment by informeti** [comment]
Alright, thanks a lot!

I'll close this issue then, as I consider it resolved with 7cdb09958bd9099938502720d7b5bf399f7d0d7b and 981d05d6f222b7983e51dde8fcc5e663a49b762e.
